File: PRACTICA PRIMER PARCIAL/Ejercicio11/Java/GenerarSecuencia.java
import java.util.ArrayList;
import java.util.List;

public class GenerarSecuencia {

    public static List<Integer> generarSecuencia(int n) {
        List<Integer> secuencia = new ArrayList<>();
        int contador = 0;

        for (int i = 0; i < n; i++) {
            if (contador == 3) {
                secuencia.add(0);
                contador = 0;
            } else {
                secuencia.add(1);
                contador++;
            }
        }

        return secuencia;
    }

    public static void main(String[] args) {
        int nTerminos = 20;
        List<Integer> secuencia = generarSecuencia(nTerminos);

        for (int num : secuencia) {
            System.out.print(num + " ");
        }
    }
}

File: PRACTICA PRIMER PARCIAL/Ejercicio11/Python/GenerarSecuencia.py
def generar_secuencia(n):
    secuencia = []
    contador = 0

    for i in range(n):
        if contador == 3:
            secuencia.append(0)
            contador = 0
        else:
            secuencia.append(1)
            contador += 1

    return secuencia

n_terminos = 20
secuencia = generar_secuencia(n_terminos)
print(secuencia)
